### ISSUE TITLE
Improve implementation of constructors

### ISSUE THREAD

**Issue by elfring** [JamesGriffin/Path-Tracer#1]
I have noticed that a couple of assignments are used in the constructor bodies.

Examples:
- [Camera](https://github.com/JamesGriffin/Path-Tracer/blob/d67fedd7cbf5056fa1927c00e208f7bed38868ca/src/camera.cpp#L4)
- [Material](https://github.com/JamesGriffin/Path-Tracer/blob/d67fedd7cbf5056fa1927c00e208f7bed38868ca/src/material.cpp#L7)
- [Sphere](https://github.com/JamesGriffin/Path-Tracer/blob/1d572953cbff01d72cdb124514e8834049353268/src/objects.cpp#L17)

The recommended way for performing efficient construction is to use the initialisation list.
- [C++ FAQ](https://isocpp.org/wiki/faq/ctors#init-lists)
- [Article by Alex Allain](http://cprogramming.com/tutorial/initialization-lists-c++.html)
- [Description by Danny Kalev](http://www.informit.com/guides/content.aspx?g=cplusplus&seqNum=15&rl=1)

**Comment by JamesGriffin** [comment]
Yep, initilisation lists are indeed the way to go. I'll switch them over when I next clean up the code.
